frontend/src/components/login/Login.tsx

```diff
@@ -37,12 +49,111 @@ export default function LoginComponent() {
 
   useEffect(() => {
     setArePasswordsEqual(!(password !== checkPassword && (password !== "" && checkPassword !== "")));
-  }, [password, checkPassword])
+    
+    if (password !== "" && checkPassword !== "" && password.length < 8) {
+      setErrorMsg("Password should contain 8 characters or more.");
+    } else if (!arePasswordsEqual) {
+      setErrorMsg("Passwords do not match. Please try again.")
+    } else {
+      setErrorMsg("");
+    }
+  }, [password, checkPassword, setPassword, setCheckPassword, arePasswordsEqual]);
+
+  useEffect(() => {
+    if (!(email.includes("@") && email.includes(".")) && email !== "") {
```

Comment:
You can use <input type="email" ... > in the form, then the browser will auto handle the email validation.